server/s3_object_data_copier.h

```diff
@@ -63,9 +62,9 @@ class S3ObjectDataCopier {
   bool write_in_progress;
   // Size of each ev buffer (e.g, 16384)
   size_t size_of_ev_buffer;
-  // Queue of ev Buffers containing read data blocks,
-  // which will be freed after write data completion
-  std::deque<struct evbuffer*> read_data_buffer;
+
+  struct evbuffer* p_evbuffer_read = nullptr;
```

Comment:
You can put coment about how p_evbuffer_read and p_evbuffer_write are related

Reply:
p_evbuffer_write holds blocks of data which are currently being written.
p_evbuffer_read contains blocks of data which has already been read and ready to write.